src/notebooks/controllers/helpers.ts

```diff
@@ -39,3 +39,12 @@ export async function createActiveInterpreterController(
         return controller;
     }
 }
+
+export async function isActiveInterpreter(
+    metadata: KernelConnectionMetadata,
+    resource: Resource,
+    interpreters: IInterpreterService
+) {
+    const pythonInterpreter = await interpreters.getActiveInterpreter(resource);
+    return pythonInterpreter?.envPath?.toString() === metadata.interpreter?.envPath?.toString();
```

Comment:
use `pythonInterpreter?.id === metadata.interpreter?.id`
That is the right way to compare interpreters (Id is provided by Python extension and is a unique identifier)